subsys/bluetooth/controller/ll_sw/ctrl.c

```diff
@@ -4820,21 +4958,29 @@ static u32_t access_addr_get(void)
 		/* It shall have no more than 24 transitions */
 		if (transitions > 24) {
 			if (consecutive_bit) {
-				access_addr &= ~((1 << (bit_idx + 1)) - 1);
+				access_addr &= ~(BIT(bit_idx + 1) - 1);
 			} else {
-				access_addr |= ((1 << (bit_idx + 1)) - 1);
+				access_addr |= (BIT(bit_idx + 1) - 1);
 			}
 
 			break;
 		}
 	}
 
-	/** @todo proper access address calculations
-	 * It shall not be the advertising channel packets Access Address.
+	/* It shall not be the advertising channel packets Access Address.
 	 * It shall not be a sequence that differs from the advertising channel
 	 * packets Access Address by only one bit.
-	 * It shall not have all four octets equal.
 	 */
+	adv_aa_check = access_addr ^ 0x8e89bed6;
+	if (util_ones_count_get((u8_t *)&adv_aa_check,
+				sizeof(adv_aa_check)) <= 1) {
+		goto again;
+	}
+
+	/* It shall not have all four octets equal. */
+	if (!((access_addr & 0xFFFF) ^ (access_addr >> 16))) {
```

Comment:
Isn't this check stricter than it needs to be? E.g. it would consider something like 0x11221122 as invalid even though its octets are not all equal.